File: chapter_1/shortestWithBaseCase.py
def shortestWithBaseCase(makeRecursiveCall):
    print(f'shortestWithRecursiveCase({makeRecursiveCall}) called.')

    if not makeRecursiveCall:
        print("Returning from base case.")
        return
    else:
        shortestWithBaseCase(False)
        print("Returning from recursive case.")
        return

if __name__ == "__main__":
    print()
    print("Calling shortestWithBaseCase(False)")
    shortestWithBaseCase(False)
    print()

    print("Calling shortestWithBaseCase(True)")
    shortestWithBaseCase(True)
    print()

File: chapter_1/shortestWithBaseCase.js
function shortestWithBaseCase(makeRecursiveCall) {
	document.write("shortestWithBaseCase(" + makeRecursiveCall + ") called.<br />");
	if (makeRecursiveCall === false) {
		document.write("Returning from base case.<br />");
		return;
	} else {
		shortestWithBaseCase(false);
		document.write("Returning from Recursive case.<br />");
		return;
	}

}


document.write("Calling shortestWithBaseCase(false):<br />");
shortestWithBaseCase(false);
document.write("<br />");

document.write("Calling shortestWithBaseCase(true):<br />");
shortestWithBaseCase(true);
